src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -113,6 +105,14 @@ func parseParams(
 	}
 	params.End = end
 
+	timeout := fetchOpts.Timeout
+	if timeout <= 0 {
+		err := fmt.Errorf("expected positive timeout, instead got: %d", timeout)
+		return params, xerrors.NewInvalidParamsError(
+			fmt.Errorf(formatErrStr, handleroptions.TimeoutParam, err))
+	}
```

Comment:
I think this should already be handled when `NewFetchOptionsBuilder.Validate()` is called before entering this function